### PR TITLE
Fix bug in parsing avro schemas

### PR DESCRIPTION
**Pull request by ghost**
**Issue**: In class AvroSchema,  schemas are not ordered properly from the ones with least dependencies to the ones with most dependencies.

**Detailed explanation**: Avro schemas in kafkasnoop are parsed in order from the ones that don’t depend on other schemas/objects, to the ones depending on the most.

Parsing them is done in this order because the Avro schema `Parser` adds every schema it parses to the names known to the `Parser`. If it needs to parse a schema with dependencies, it checks in the known schemas to resolve the dependencies. Incorrect ordering leads to a lot of schemas not being parsed.

### CODE REVIEW

File: lib/avro/src/main/kotlin/kafkasnoop/avro/AvroSchema.kt [hunk 1]
```diff
@@ -42,5 +43,31 @@ data class AvroSchema(val fullName: String, val needs: List<String>, val schema:
...
         }
+
+        private fun extractNonPrimitiveTypes(elem: JsonElement?): List<String> {
+
+            if (elem == null) {
+                return emptyList()
+            }
+
+            return if (elem.isJsonObject) {
+                extractNonPrimitiveTypes(elem.asJsonObject.get("type"))
```
Comment (abellagonzalo) [inline lib/avro/src/main/kotlin/kafkasnoop/avro/AvroSchema.kt]: The return value of these function is getting lost. Is that the expected behaviour? Or do you mean:
```
extractNonPrimitiveTypes(elem.asJsonObject.get("type")) + extractNonPrimitiveTypes(elem.asJsonObject.get("items"))
```

File: lib/avro/src/main/kotlin/kafkasnoop/avro/AvroSchema.kt [hunk 2]
```diff
@@ -42,5 +43,31 @@ data class AvroSchema(val fullName: String, val needs: List<String>, val schema:
...
+                extractNonPrimitiveTypes(elem.asJsonObject.get("items"))
+            } else if (elem.isJsonArray) {
+                elem.asJsonArray.filter {
+                    if (it.isJsonPrimitive) {
```
Comment (abellagonzalo) [inline lib/avro/src/main/kotlin/kafkasnoop/avro/AvroSchema.kt]: your can simplify this with `.filter { it.isJsonPrimitive && it.asString !in primitive }`

**Review by abellagonzalo — CHANGES_REQUESTED** [review]
Are there any tests?

File: lib/avro/src/main/kotlin/kafkasnoop/avro/AvroSchema.kt (hunk 1 above)
Comment (ghost) [reply lib/avro/src/main/kotlin/kafkasnoop/avro/AvroSchema.kt]: Yeah, you're right, I've missed this. I'll update it. Thanks!

**Comment by ghost** [comment]
> Are there any tests?

I'll see how to write additional tests

**Review by abellagonzalo — APPROVED** [review]

**Pull request merged**